contracts/contracts/ResolutionRegistry.sol

```diff
@@ -4,16 +4,43 @@ pragma solidity ^0.8.12;
 import "./CrossDomainRestrictedCalls.sol";
 import "./BeamerUtils.sol";
 
+/// The resolution registry.
+///
+/// This contract resides on the source L2 chain and is tasked with keeping track of results
+/// of L1 resolution. In particular, it stores the information about known fills and fillers,
+/// as well as fills that were marked invalid. This information is used by the :sol:contract:`RequestManager`
+/// to resolve claims.
+///
+/// .. note::
+///
+///   This contract can only be called by the :sol:contract:`Resolver` contract, via a
+///   chain-dependent messenger contract.
 contract ResolutionRegistry is CrossDomainRestrictedCalls {
+    /// Emitted when a request has been resolved via L1 resolution.
+    ///
+    /// .. seealso:: :sol:func:`resolveRequest`
     event RequestResolved(bytes32 requestHash, address filler, bytes32 fillId);
 
+    /// Emitted when a fill hash has been invalidated.
+    ///
+    /// .. seealso:: :sol:func:`invalidateFillHash`
     event FillHashInvalidated(bytes32 fillHash);
 
-    // mapping from requestHash to FillInfo
+    /// Maps request hashes to fill infos.
     mapping(bytes32 => BeamerUtils.FillInfo) public fillers;
-    // mapping from fillHash to validity flag
+
+    /// The set of invalid fill hashes.
     mapping(bytes32 => bool) public invalidFillHashes;
 
+    /// Resolve the specified request.
+    ///
+    /// Callable only by the :sol:contract:`Resolver` contract via a messenger, this
```

Comment:
Technically, it is only callable by the `l1_messenger_contract`. I know that this contract can only be called by the resolver, but in the l1 messenger contract you also describe that the messenger can only be called by the resolver. In order to stay consistent how the message chain is described throughout the whole documentation, I would actually prefer to make it clear that the function is only allowed to be called by the l1 messenger

Reply:
> Technically, it is only callable by the `l1_messenger_contract`. I know that this contract can only be called by the resolver, but in the resolver you also describe that it can only be called by the proof submitter. So ultimately the addresses which are stored as `callers`. In order to stay consistent how the message chain is described throughout the whole documentation, I would actually prefer to make it clear that the function is only allowed to be called by the l1 messenger

Hm, I thought I was already being consistent -- by including the "... via a messenger ..." wording everywhere.
Technically, yes, the messengers are the immediate neighbours in the call chain, however, in the grand scheme
of things I feel it's more important to highlight where the call _really_ comes from (i.e. the proof submitter, or the resolver).

Reply:
I thought about a bit more, how could one argue that in the grand scheme the resolver the *real caller*. One could argue that the *real caller* is defined where the call is constructed. 

In the IProofSubmitter the call to the resolver is constructed (`abi.encode...`). 
In the same sense the resovler contract constructs the resolution registry call. Following this argumentation, your documentation makes sense. 

I always try to read the docs as if I was an external developer getting familiar with our codebase. Reading in the docs that we have `Resolver -> Messenger -> ResolutionRegistry` and then reading in the contract documentation that the Resolver is the caller of the ResolutionRegistry would confuse me and I really want to make sure that we can prevent this confusion.
Defining the caller as the one who constructs the call would separate the definition of the "real caller" from the technical message flow. I think it is important to explicitly distiguish to not confuse readers.

Reply:
I updated the wording.